### ISSUE TITLE
optimize image load copying

### ISSUE THREAD

**Issue by medyagh** [kubernetes/minikube#6393]
for example for kic:

we do chmod and mkdir and each one is taking 1s and also they are not being done in paralell !

```
I0124 13:34:44.677706   18965 cache_images.go:172] Loading image from cache: /Users/medya/.minikube/cache/images/k8s.gcr.io/kube-scheduler_v1.17.0
I0124 13:34:44.904755   18965 kic_runner.go:115] Done: [docker exec --privileged minikube mkdir -p /var/lib/minikube/images]: (2.785642328s)
I0124 13:34:45.006597   18965 kic_runner.go:115] Done: [docker exec --privileged minikube mkdir -p /var/lib/minikube/images]: (3.111271986s)
I0124 13:34:45.129123   18965 cache_images.go:86] "gcr.io/k8s-minikube/storage-provisioner:v1.8.1" needs transfer: "gcr.io/k8s-minikube/storage-provisioner:v1.8.1" does not exist at hash "4689081edb103a9e8174bf23a255bfbe0b2d9ed82edc907abab6989d1c60f02c" in container runtime
I0124 13:34:45.129205   18965 cache_images.go:172] Loading image from cache: /Users/medya/.minikube/cache/images/gcr.io/k8s-minikube/storage-provisioner_v1.8.1
I0124 13:34:45.254320   18965 cache_images.go:86] "kubernetesui/metrics-scraper:v1.0.2" needs transfer: "kubernetesui/metrics-scraper:v1.0.2" does not exist at hash "3b08661dc379d9f80155be9d658f71578988640357ebae1aab287d6954c723d1" in container runtime
I0124 13:34:45.254370   18965 cache_images.go:172] Loading image from cache: /Users/medya/.minikube/cache/images/kubernetesui/metrics-scraper_v1.0.2
I0124 13:34:45.361082   18965 cache_images.go:86] "kubernetesui/dashboard:v2.0.0-beta8" needs transfer: "kubernetesui/dashboard:v2.0.0-beta8" does not exist at hash "eb51a359752560a66f314602e87155b75f428fb838bf951079ff1f9621958c0c" in container runtime
I0124 13:34:45.361127   18965 cache_images.go:172] Loading image from cache: /Users/medya/.minikube/cache/images/kubernetesui/dashboard_v2.0.0-beta8
I0124 13:34:45.459200   18965 kic_runner.go:115] Done: [docker exec --privileged minikube mkdir -p /var/lib/minikube/images]: (2.566192447s)
I0124 13:34:45.585583   18965 kic_runner.go:115] Done: [docker exec --privileged minikube mkdir -p /var/lib/minikube/images]: (1.745791575s)
I0124 13:34:45.796019   18965 kic_runner.go:115] Done: [docker exec --privileged minikube mkdir -p /var/lib/minikube/images]: (1.729713581s)
I0124 13:34:46.003513   18965 kic_runner.go:115] Done: [docker exec --privileged minikube mkdir -p /var/lib/minikube/images]: (1.61728111s)
I0124 13:34:46.111368   18965 kic_runner.go:115] Done: [docker exec --privileged minikube mkdir -p /var/lib/minikube/images]: (1.433337166s)
I0124 13:34:58.240709   18965 kic_runner.go:115] Done: [docker exec --privileged minikube mkdir -p /var/lib/minikube/images]: (13.111081791s)
I0124 13:34:59.265356   18965 kic_runner.go:115] Done: [docker exec --privileged minikube mkdir -p /var/lib/minikube/images]: (13.903986485s)
I0124 13:34:59.888966   18965 kic_runner.go:115] Done: [docker exec --privileged minikube mkdir -p /var/lib/minikube/images]: (14.634392743s)
I0124 13:35:00.144223   18965 kic_runner.go:115] Done: [docker exec --privileged minikube sudo chmod 0644 /var/lib/minikube/images/coredns_1.6.5]: (13.847820227s)
I0124 13:35:00.144305   18965 containerd.go:235] Loading image: /var/lib/minikube/images/coredns_1.6.5
I0124 13:35:00.230636   18965 kic_runner.go:115] Done: [docker exec --privileged minikube sudo chmod 0644 /var/lib/minikube/images/etcd_3.4.3-0]: (11.014143027s)
I0124 13:35:00.328642   18965 kic_runner.go:115] Done: [docker exec --privileged minikube sudo chmod 0644 /var/lib/minikube/images/kube-apiserver_v1.17.0]: (7.367570832s)
I0124 13:35:00.480488   18965 kic_runner.go:115] Done: [docker exec --privileged minikube sudo chmod 0644 /var/lib/minikube/images/kube-proxy_v1.17.0]: (5.784794193s)
I0124 13:35:00.655047   18965 kic_runner.go:115] Done: [docker exec --privileged minikube sudo chmod 0644 /var/lib/minikube/images/pause_3.1]: (9.472745642s)
I0124 13:35:00.842488   18965 kic_runner.go:115] Done: [docker exec --privileged minikube sudo chmod 0644 /var/lib/minikube/images/kube-scheduler_v1.17.0]: (2.983968182s)
I0124 13:35:01.152930   18965 kic_runner.go:115] Done: [docker exec --privileged minikube sudo chmod 0644 /var/lib/minikube/images/kube-controller-manager_v1.17.0]: (4.618601407s)
I0124 13:35:03.321377   18965 kic_runner.go:115] Done: [docker exec --privileged minikube sudo chmod 0644 /var/lib/minikube/images/storage-provisioner_v1.8.1]: (3.612574899s)
I0124 13:35:03.749888   18965 kic_runner.go:115] Done: [docker exec --privileged minikube sudo chmod 0644 /var/lib/minikube/images/dashboard_v2.0.0-beta8]: (1.744002973s)
I0124 13:35:04.510107   18965 kic_runner.go:115] Done: [docker exec --privileged minikube sudo ctr -n=k8s.io images import /var/lib/minikube/images/coredns_1.6.5]: (4.365638975s)
I0124 13:35:04.510151   18965 cache_images.go:194] Transferred and loaded /Users/medya/.minikube/cache/images/k8s.gcr.io/coredns_1.6.5 from cache
I0124 13:35:04.510178   18965 containerd.go:235] Loading image: /var/lib/minikube/images/etcd_3.4.3-0
I0124 13:35:11.209669   18965 kic_runner.go:115] Done: [docker exec --privileged minikube sudo ctr -n=k8s.io images import /var/lib/minikube/images/etcd_3.4.3-0]: (6.698740575s)
I0124 13:35:11.209731   18965 cache_images.go:194] Transferred and loaded /Users/medya/.minikube/cache/images/k8s.gcr.io/etcd_3.4.3-0 from cache
I0124 13:35:11.209762   18965 containerd.go:235] Loading image: /var/lib/minikube/images/kube-apiserver_v1.17.0
I0124 13:35:15.971622   18965 kic_runner.go:115] Done: [docker exec --privileged minikube sudo ctr -n=k8s.io images import /var/lib/minikube/images/kube-apiserver_v1.17.0]: (4.76145704s)
I0124 13:35:15.971700   18965 cache_images.go:194] Transferred and loaded /Users/medya/.minikube/cache/images/k8s.gcr.io/kube-apiserver_v1.17.0 from cache
I0124 13:35:15.971726   18965 containerd.go:235] Loading image: /var/lib/minikube/images/kube-proxy_v
```

**Comment by afbjorklund** [comment]
chmod/mkdir: I think that @tstromberg addressed this in another issue: ~#6385~

We _have_ to load the images one-by-one, or else we risk running out of memory.
I think it is mostly I/O bound anyway - not sure if we gain anything by parallel ?

Added a mutex in 98a2a557c584ee5eeae9355a2922bd8fd9da4a48, not needed for Docker since the daemon has one.

**Comment by tstromberg** [comment]
Closing because I believe this is solved and/or a red herring. Let's reopen if we see something more specific.
